naga/src/proc/constant_evaluator.rs

```diff
@@ -2186,6 +2190,31 @@ impl<'a> ConstantEvaluator<'a> {
         }
     }
 
+    /// Returns the total number of components, after flattening, of a vector compose expression.
+    fn vector_compose_flattened_size(
+        &self,
+        components: &[Handle<Expression>],
+    ) -> Result<usize, ConstantEvaluatorError> {
+        components
+            .iter()
+            .try_fold(0, |acc, c| -> Result<_, ConstantEvaluatorError> {
+                let ty = self.resolve_type(*c)?;
+                let ty = match ty {
+                    super::TypeResolution::Handle(handle) => &self.types[handle].inner,
+                    super::TypeResolution::Value(ref inner) => inner,
+                };
+                let size = match *ty {
```

Comment:
```suggestion
                let size = match *ty.inner_with(&self.types) {
```

Reply:
I knew there was shortcut for this but my editor's autocomplete was letting me down :slightly_smiling_face: